### ISSUE TITLE
FaceExpressionNet tflite

### ISSUE THREAD

**Issue by andBabaev** [justadudewhohacks/face-api.js#803]
Hi
I want to convert an emotion recognition model to tflite. But after conversion, the model works much worse. She almost always predicts "neutral", "angry" or "happy". The model from face api demo works much better. 

I created a model architecture using tensorflow.keras (tf==2.3.1). Then I loaded the weights using [decode_weights](https://github.com/tensorflow/tfjs/blob/77c4ef10fd3d8458f6958a52bac92fdaae09b885/tfjs-converter/python/tensorflowjs/read_weights.py#L126). Then saved as tf SavedModel. Then I converted to tflite

Code for creating architecture 
```
from tensorflow.keras import layers, Model

def get_face_exp_net():
    inputs = layers.Input((112,112,3))

    out1 = layers.ReLU()(layers.Conv2D(32, 3, strides=[2,2], padding='same')(inputs))
    out2 = layers.SeparableConv2D(32, 3, padding='same')(out1)
    in3 = layers.ReLU()(layers.Add()([out1, out2]))
    out3 = layers.SeparableConv2D(32, 3, padding='same')(in3)
    in4 = layers.ReLU()(layers.Add()([out1, out2, out3]))
    out4 = layers.SeparableConv2D(32, 3, padding='same')(in4)
    end = layers.ReLU()(layers.Add()([out1, out2, out3, out4]))

    out1 = layers.ReLU()(layers.SeparableConv2D(64, 3, strides=[2,2], padding='same')(end))
    out2 = layers.SeparableConv2D(64, 3, padding='same')(out1)
    in3 = layers.ReLU()(layers.Add()([out1, out2]))
    out3 = layers.SeparableConv2D(64, 3, padding='same')(in3)
    in4 = layers.ReLU()(layers.Add()([out1, out2, out3]))
    out4 = layers.SeparableConv2D(64, 3, padding='same')(in4)
    end = layers.ReLU()(layers.Add()([out1, out2, out3, out4]))

    out1 = layers.ReLU()(layers.SeparableConv2D(128, 3, strides=[2,2], padding='same')(end))
    out2 = layers.SeparableConv2D(128, 3, padding='same')(out1)
    in3 = layers.ReLU()(layers.Add()([out1, out2]))
    out3 = layers.SeparableConv2D(128, 3, padding='same')(in3)
    in4 = layers.ReLU()(layers.Add()([out1, out2, out3]))
    out4 = layers.SeparableConv2D(128, 3, padding='same')(in4)
    end = layers.ReLU()(layers.Add()([out1, out2, out3, out4]))

    out1 = layers.ReLU()(layers.SeparableConv2D(256, 3, strides=[2,2], padding='same')(end))
    out2 = layers.SeparableConv2D(256, 3, padding='same')(out1)
    in3 = layers.ReLU()(layers.Add()([out1, out2]))
    out3 = layers.SeparableConv2D(256, 3, padding='same')(in3)
    in4 = layers.ReLU()(layers.Add()([out1, out2, out3]))
    out4 = layers.SeparableConv2D(256, 3, padding='same')(in4)
    end = layers.ReLU()(layers.Add()([out1, out2, out3, out4]))

    end = layers.AvgPool2D((7,7), strides=(2,2), padding='valid')(end)
    end = layers.Flatten()(end)
    # end = layers.Dropout(0.5)(end)
    end = layers.Dense(7)(end)
    end = layers.Softmax()(end)

    return Model(inputs=inputs, outputs=end)
```

Code for loading of weights and converting to tflite
```
    model_config = {'func': get_face_exp_net(),
                     'name': 'face expression net',
                     'manifest': 'models/tfjs_models/face_expression_model-weights_manifest.json',
                     'weights': 'models/tfjs_models/face_expression_model-shard1'
                     }
    print('Converting of ', model_config['name'])

    with open(model_config['manifest']) as json_file:
        manifest = json.load(json_file)

    weights_bytes = read_weights(model_config['weights'])
    # [{'name':'layer_name1', 'data':np.array with weights}, 
    # {'name':'layer_name2', 'data':np.array with weights},
    # ...]
    weights = tfjs.read_weights.decode_weights([manifest], [weights_bytes])[0]
    
    model = model_config['func']
    target_layers_ids = []
    for i, l in enumerate(model.layers):
        if 'conv' in l.name or 'dense' in l.name:
            # print(i, l.name, [w.shape for w in l.get_weights()])
            target_layers_ids.append(i)

    # set weights to model
    w_counter = 0
    for layer_id in target_layers_ids:
        num_w = len(model.layers[layer_id].get_weights())
        w_list = [w['data'] for w in weights[w_counter:w_counter+num_w]]
        model.layers[layer_id].set_weights(w_list)
        w_counter+=num_w

    model_name = "_".join(model_config['name'].split())
    tf_model_path = 'models/tf_models/{}'.format(model_name)
    model.save(tf_model_path)
    print('TF model is saved to ', tf_model_path)

    converter = tf.lite.TFLiteConverter.from_saved_model(tf_model_path)
    tflite_model = converter.convert()
    tflite_model_path = 'models/tflite_models/{}.tflite'.format(model_name)
    open(tflite_model_path, "wb").write(tflite_model)
    print('TFLite model is saved to ', tflite_model_path)
```

My code for inference
```
class FaceExpressionModel:
    def __init__(self, tflite_path=os.getcwd()+'/tflite_face_api/models/tflite_models/face_expression_net.tflite',
                ):

        self.model = tf.lite.Interpreter(model_path=tflite_path)
        self.model.allocate_tensors()
        self.input_details = self.model.get_input_details()
        self.output_details = self.model.get_output_details()
        self.image_size = self.input_details[0]['shape'][1:3] # [112,112]

    def normalize(self, img):
        img[..., 0] -= 122.782
        img[..., 1] -= 117.001
        img[..., 2] -= 104.298

        return img / 255.0

    def pad_to_square(self, img):
        h, w, ch = img.shape

        if h==w:
            return img

        max_dim = max(h, w)
        dim_diff = int(np.abs(h - w) * 0.5)
        pad_array = np.zeros((max_dim, max_dim, ch))
        if h > w:
            pad_array[:, dim_diff:dim_diff+w, :] = img
        else:
            pad_array[dim_diff:dim_diff+h, :, :] = img
        return pad_array

    def preprocessing(self, img):
        img = cv2.resize(img, (self.image_size)).astype(np.float32)
        img = self.pad_to_square(img)
        img = self.normalize(img)
        return np.expand_dims(img, axis = 0)

    def predict(self, img):
        """
        Does inference, preprocessing and returns probabilities for emotions

        Arguments:
        ----------
        img : 3d np.ndarray
            Single rgb image

        Return:
        ---------
        1d np.ndarray
            array of class probabilites, values from 0 to 1
        """
        img = self.preprocessing(img)
        self.model.set_tensor(self.input_details[0]['index'], img)  
        self.model.invoke()
        preds = self.model.get_tensor(self.output_details[0]['index'])[0]
        return preds
```

**Comment by vladmandic** [comment]
assuming everything is correct, likely issue is loss of resolution or clipping of values - `tflite` conversion quantizes models to `uint8` which may not be enough for all interim processing results.
